### DISCUSSION TITLE
MEGA65 support for ca65 only

### DISCUSSION THREAD

**Discussion by dillof** [cc65/cc65#1782] [General]
hi,

I'm in the process of adding MEGA65 (and 45GS02) support to ca65. However, I'm not interested in the C compiler or standard library. Would such a change be acceptable?

My fork is at https://github.com/dillof/cc65

So far, I've added MEGA65 to the list of supported targets and added the minimal stubs to the standard library to make the build succeed. It works when only using assembler files.

Next I'll add support for the compound instructions of the 45GS02 and some asminc files with register definitions.

thanks,
dillo

**Comment by mrdudz** [comment]
You shouldnt have to touch anything near the compiler or libraries to add a new CPU to ca65 - see how 65816 support was done

**Comment by dillof** [comment]
I'm not just adding a new CPU, I'm also adding a new target (`-t mega65`) with its own define (`__MEGA65__`). I can see if I can add this to ca65/cl65 only, without touching cc65 or the libraries.

**Comment by mrdudz** [comment]
I see, that stuff is global i think - please no special cases to exclude it from the compiler, that will only make it harder to extend the compiler at some point.

**Comment by groessler** [comment]
On 6/29/22 14:07, Dieter Baron wrote:
>
> I'm not just adding a new CPU, I'm also adding a new target (|-t 
> mega65|) with its own define (|__MEGA65__|). I can see if I can add 
> this to ca65/cl65 only, without touching cc65 or the libraries.
>

What's the reason for such a target?

**Reply by dillof** [reply]
> On 6/29/22 14:07, Dieter Baron wrote: I'm not just adding a new CPU, I'm also adding a new target (|-t mega65|) with its own define (|__MEGA65__|). I can see if I can add this to ca65/cl65 only, without touching cc65 or the libraries.
> What's the reason for such a target?

All other computers have their own target, so why not the MEGA65? It is useful for conditional compilation of sections specifically for the MEGA65; I'm using this quite a lot in my programs.
